be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -27,15 +31,17 @@ case object RollCallHandler extends MessageHandler {
     case graphMessage@_ => graphMessage
   }
 
-  def handleCreateRollCall(rpcMessage: JsonRpcRequest): GraphMessage =
-    rpcMessage.getParamsChannel.decodeSubChannel match {
-      case Some(_) => dbAskWritePropagate(rpcMessage)
-      case _ => Right(PipelineError(
-        ErrorCodes.INVALID_DATA.id,
-        s"Unable to create meeting: invalid encoded laoId '${rpcMessage.getParamsChannel}'",
-        rpcMessage.id
-      ))
+  def handleCreateRollCall(rpcMessage: JsonRpcRequest): GraphMessage = {
+    val message: Message = rpcMessage.getParamsMessage.get
+
+    val f: Future[GraphMessage] = (dbActor ? DbActor.Write(rpcMessage.getParamsChannel, message)).map {
+      case DbActor.DbActorWriteAck => Left(rpcMessage)
+      case DbActor.DbActorNAck(code, description) => Right(PipelineError(code, description, rpcMessage.id))
+      case _ => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, "Database actor returned an unknown answer", rpcMessage.id))
     }
+
+    Await.result(f, duration)
```

Comment:
is there any form of timeout on this await?

Reply:
The future fails if `duration` is exceeded :)